### PR TITLE
various changes

### PR DESCRIPTION
**Pull request by DavidBJaffe**
- Document cross filtering, foursie filtering and weak chain filtering on heuristics page.
- Fix infinite loop in help printing (would never be seen by user).
- Tweak computations of features.
- Tweak the help text for the lvar "filter".
- Make IMGT arg work with MOUSE (internal).
- Detect defective reference sequences and abort, allowing for override.
  Note that we are granting a "free pass" to four reference sequences of our own that are broken.
  We should separately correct those, and eventually remove the free pass.
- Fix some things in the summary:
   * remove duplicate cell count
   * refer to e.g. B-B doublet rate and not just doublet rate
   * don't say "heavy chain / TRB" etc. -- just say which one applies.
- Fix ancient bugs that are very rarely manifested:
   * Causing amino acid display to be truncated (and asserting).
   * When CDR3 in json input file has an asterisk in it.
   * When CDR3 in json input file is too long. 
- Major change to filtering.  This drops half of the graph filtering, which in some cases greatly
reduces the number of cells that are deleted.  Turning off this filtering results in one additional
multi-donor clonotype (FP) in the large specificity test, but the additional FP seems like a consequence
of the core algorithm, so we should probably see it.
- New tests:
   * The ~400 dataset test now locks the number of cells in clonotypes, so that it cannot accidentally change.  We also lock other metrics from that test.
   * That test also is now run in such a way as to generate info for all clonotypes, even though it isn't 
printed.  We have found three bugs this way (as described above).
   * We test the output displayed in the annotated example to make sure that it hasn't changed.
- Test results quoted on the site are now locked to the actual test results so they don't get out of date.
- Give up on updating the annotated example and state on the site that we don't keep it exactly current.
- Add argument BUILT_IN to force use of the built in reference (for human or mouse).
- Add internal option NALL_GEX to turn off all default filters except the GEX filter.
- Add internal option REPROD to accept nonproductive contigs, recompute productive, then reject nonproductive.
- Generate a summary table for GEX clusters (internal only).

### CODE REVIEW

**Comment by meryllewis** [comment]
jenkins retest this please

File: enclone/src/UNDOC_OPTIONS [hunk 1]
```diff
@@ -115,11 +112,12 @@ LVARS entries
 - npe<n> = total number of cells in this clonotype that are within PCA distance n of this cell
 - ppe<n> = percent of all gex cells within PCA distance n of this cell that are in this clonotype
 
-IMGT: specify the IMGT reference, only implemented for human, forces RE so slow
+IMGT: specify the IMGT reference, forces RE so slow
 How this reference was created
 - go into a new directory
 - /mnt/home/jaffe/repos/cellranger/lib/bin/fetch-imgt --genome human-imgt-20200415
 - mkvdjref --genome=vdj_IMGT_human_20200415 --seqs=human-imgt-20200415-mkvdjref-input.fasta
+- alse for mouse
```
Comment (wyattmcdonnell) [inline enclone/src/UNDOC_OPTIONS:120]:
```suggestion
- also for mouse
```

File: enclone_main/src/main_enclone.rs [hunk 2]
```diff
@@ -648,26 +650,40 @@ pub fn main_enclone(args: &Vec<String>) {
...
-                if ctl.gen_opt.imgt_fix {
-                    // Fix IGHJ6.
-                    if s == "ATTACTACTACTACTACGGTATGGACGTCTGGGGCCAAGGGACCACGGTCACCGTCTCCTCA"
-                        .to_string()
-                        || s == "ATTACTACTACTACTACTACATGGACGTCTGGGGCAAAGGGACCACGGTCACCGTCTCCTCA"
+            if !ctl.gen_opt.mouse {
+                let imgt =
+                    "/mnt/opt/refdata_cellranger/vdj/vdj_IMGT_human_20200415-0.0.0/fasta/regions.fa";
+                if ctl.gen_opt.descrip {
+                    println!("using imgt human reference");
```
Comment (wyattmcdonnell) [inline enclone_main/src/main_enclone.rs:657]:
```suggestion
                    println!("using IMGT human reference");
```

File: enclone_main/src/main_enclone.rs [hunk 3]
```diff
@@ -648,26 +650,40 @@ pub fn main_enclone(args: &Vec<String>) {
...
+                        }
                     }
+                    refx += &s;
+                    refx += &"\n";
+                }
+            } else {
+                let imgt =
+                    "/mnt/opt/refdata_cellranger/vdj/vdj_IMGT_mouse_20180723-2.2.0/fasta/regions.fa";
+                if ctl.gen_opt.descrip {
+                    println!("using imgt mouse reference");
```
Comment (wyattmcdonnell) [inline enclone_main/src/main_enclone.rs:679]:
```suggestion
                    println!("using IMGT mouse reference");
```

File: enclone_main/src/main_enclone.rs [hunk 4]
```diff
@@ -738,6 +762,249 @@ pub fn main_enclone(args: &Vec<String>) {
         to_ref_index.insert(refdata.id[i] as usize, i);
     }
 
+    // Flag defective reference sequences.
+
+    let mut log = Vec::<u8>::new();
+    let mut count = 0;
+    let mut broken = vec![false; refdata.refs.len()];
+    for i in 0..refdata.refs.len() {
+        // Determine chain type and exclude thosse other than IGH, IGK, IGL, TRA and TRB.
```
Comment (wyattmcdonnell) [inline enclone_main/src/main_enclone.rs:771]:
```suggestion
        // Determine chain type and exclude those other than IGH, IGK, IGL, TRA and TRB.
```

**Review by wyattmcdonnell — APPROVED** [review]
Good to go! 🛫

**Pull request merged**